src/Xamarin.Android.Build.Tasks/Tasks/GetImportedLibraries.cs

```diff
@@ -35,22 +35,37 @@ public class GetImportedLibraries : AndroidTask
 
 		public override bool RunTask ()
 		{
-			if (!Directory.Exists (TargetDirectory)) {
-				Log.LogDebugMessage ("Target directory was not found");
-				return true;
-			}
-
 			var manifestDocuments = new List<ITaskItem> ();
 			var nativeLibraries   = new List<ITaskItem> ();
 			var jarFiles          = new List<ITaskItem> ();
-			foreach (var file in Directory.EnumerateFiles (TargetDirectory, "*", SearchOption.AllDirectories)) {
-				if (file.EndsWith (".so", StringComparison.OrdinalIgnoreCase)) {
-					if (AndroidRidAbiHelper.GetNativeLibraryAbi (file) != null)
-						nativeLibraries.Add (new TaskItem (file));
-				} else if (file.EndsWith (".jar", StringComparison.OrdinalIgnoreCase)) {
-					jarFiles.Add (new TaskItem (file));
-				} else if (file.EndsWith (".xml", StringComparison.OrdinalIgnoreCase)) {
-					if (Path.GetFileName (file) == "AndroidManifest.xml") {
+			foreach (var extractedDirectory in ExtractedDirectories) {
+				if (!Directory.Exists (extractedDirectory.ItemSpec)) {
+					continue;
+				}
+				string originalFile = extractedDirectory.GetMetadata (ResolveLibraryProjectImports.OriginalFile);
+				string nuGetPackageId = extractedDirectory.GetMetadata (ResolveLibraryProjectImports.NuGetPackageId);
+				string nuGetPackageVersion = extractedDirectory.GetMetadata (ResolveLibraryProjectImports.NuGetPackageVersion);
+				foreach (var file in Directory.EnumerateFiles (extractedDirectory.ItemSpec, "*", SearchOption.AllDirectories)) {
+					if (file.EndsWith (".so", StringComparison.OrdinalIgnoreCase)) {
+						if (AndroidRidAbiHelper.GetNativeLibraryAbi (file) != null)
+							nativeLibraries.Add (new TaskItem (file, new Dictionary<string, string> {
+								{ ResolveLibraryProjectImports.OriginalFile, originalFile },
```

Comment:
I would prefer that we use C#6 dictionary initializers here and elsewhere:

```csharp
nativeLibraries.Add (new TaskItem (file, new Dictionary<string, string> {
	[ResolveLibraryProjectImports.OriginalFile] = originalFile,
	[ResolveLibraryProjectImports.NuGetPackageId] = nuGetPackageId,
	[ResolveLibraryProjectImports.NuGetPackageVersion] = nuGetPackageVersion,
}));

```

Reply:
didn't even know that would work like that 🤣